doc/3/controllers/document/update/index.md

```diff
@@ -0,0 +1,67 @@
+---
+code: true
+type: page
+title: update
+description: Updates a document
+---
+
+# update
+
+Updates a document.
+
+---
+
+## Arguments
+ 
+```java
+public CompletableFuture<ConcurrentHashMap<String, Object>> update(
+      final String index,
+      final String collection,
+      final String id,
+      final ConcurrentHashMap<String, Object> document)
+throws NotConnectedException, InternalException
+
+public CompletableFuture<ConcurrentHashMap<String, Object>> update(
+      final String index,
+      final String collection,
+      final String id,
+      final ConcurrentHashMap<String, Object> document,
+      final DocumentOptions options)
```

Comment:
I disagree with that name: that option class can only be passed to methods having a use for all its exposed options, namely waitForRefresh, retryOnConflict and source. These last 2 in particular are only relevant for update (and possibly mUpdate) methods.

You have to design 1 option class per "use" (in term of functionalities), meaning that all options must have meaning for a particular method. If you use a "DocumentOption" class for methods having no use for the retryOnConflict option for instance, then this can be hard for users to understand.

Rename this to `UpdateOptions`, `DocumentUpdateOptions`, or whatever is more fitting for this method.

Reply:
I just saw that DocumentOptions already existed prior to this PR, and contains an `id` option for methods such as `document.create`. This makes my point even more relevant: there is no use for that id option, and this can even be **very** confusing for users, as the update method already takes an "id" argument.

Reply:
Changed :+1: